volume/store/store_windows.go

```diff
@@ -0,0 +1,12 @@
+package store
+
+import "strings"
+
+// normaliseVolumeName is a platform specific function to normalise the name
+// of a volume. On Windows, as NTFS is case insensitive, under
+// c:\ProgramData\Docker\Volumes\, the folders John and john would be synonymous.
+// Hence we can't allow the volume "John" and "john" to be created as seperate
+// volumes.
+func normaliseVolumeName(name string) string {
```

Comment:
@calavera Summarising the IRC conversation with @cpuguy83 

The problem is that 'store' names are case insensitive, but 'driver' names might be case sensitive. There's no commonality between the two.

One thought was to allow the driver to change the name during the stores call to vd.Create, and use that in the map of store names. However, that's slightly flawed for two reasons: One, the store.Create call makes an early call to see if it's a dup, long before it's called into the driver to get a normalised name. Second, it changes the contract between the store and drivers allowing drivers to change the name.

Another thought is to add another driver method itself to normalise the name. This has broader ramifications due to there being several out of box drivers, not in the docker code base. These would be broken by this change.

Or, we could leave it as is, and what this means is that on Linux, volume names are case insensitive, but on Windows, volume names are case sensitive. This is regardless of driver.

As to where it ended on IRC: @cpuguy83 said "I'm ok if calavera is ok" going with the last one, or being case sensitive on Windows. This is what's currently implemented in the PR.